src/jit/assertionprop.cpp

```diff
@@ -2494,32 +2471,30 @@ GenTree* Compiler::optVNConstantPropOnTree(BasicBlock* block, GenTree* tree)
 
                     case TYP_DOUBLE:
                         // Same sized reinterpretation of bits to double
-                        newTree = optPrepareTreeForReplacement(tree, tree);
-                        tree->ChangeOperConst(GT_CNS_DBL);
-                        tree->gtDblCon.gtDconVal = *(reinterpret_cast<double*>(&value));
-                        tree->gtVNPair           = ValueNumPair(vnLib, vnCns);
+                        conValTree = gtNewDconNode(*(reinterpret_cast<double*>(&value)));
                         break;
 
                     default:
-                        return nullptr;
+                        // Do not support such optimization.
+                        break;
                 }
             }
         }
         break;
 
         case TYP_REF:
-            if (tree->TypeGet() != TYP_REF)
+        {
+            assert(vnStore->ConstantValue<size_t>(vnCns) == 0);
+            if (tree->TypeGet() == TYP_REF)
             {
-                return nullptr;
+                conValTree = gtNewIconNode(0, TYP_REF);
             }
-
-            assert(vnStore->ConstantValue<size_t>(vnCns) == 0);
-            newTree = optPrepareTreeForReplacement(tree, tree);
-            tree->ChangeOperConst(GT_CNS_INT);
-            tree->gtIntCon.gtIconVal = 0;
-            tree->ClearIconHandleMask();
-            tree->gtVNPair = ValueNumPair(vnLib, vnCns);
-            break;
+            else
+            {
+                // Do not support such optimization (e.g byref(ref(0)).
+            }
+        }
```

Comment:
Minor suggestion - I think it would be cleaner, and just as clear, to move this comment prior to the `if`, and eliminate the `else` clause

Reply:
Thank you, changed.